libsplinter/src/node_registry/unified.rs

```diff
@@ -0,0 +1,359 @@
+// Copyright 2019 Cargill Incorporated
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Unified NodeRegistry implementations.
+//!
+//! This module provides a unified node registry which combines the node data from one or more
+//! read-only node registries with one local read-write node registry.  The data is merged from the
+//! local source into values from the read-only sources, allowing for an overlay or extension to
+//! existing node metadata.
+//!
+//! This module is behind the `"node-registry-unified"` feature, and is considered experimental.
+
+use std::sync::Arc;
+
+use super::{
+    MetadataPredicate, Node, NodeRegistryError, NodeRegistryReader, NodeRegistryWriter,
+    RwNodeRegistry,
+};
+
+/// Unifies a set of read-only node registries with a local, read-write node registry.
+///
+/// Nodes read from the unified registry utilize the read-only sources to fetch node definitions
+/// and then overlay any metadata values based on the local source.
```

Comment:
Not sure this description is accurate anymore.

Reply:
Updated